yt_dlp/extractor/floatplane.py

```diff
@@ -17,8 +17,107 @@
 from ..utils.traversal import traverse_obj
 
 
-class FloatplaneIE(InfoExtractor):
+class FloatplaneBaseIE(InfoExtractor):
+    def _base_extract(self, url, base_url, do_impersonate):
```

Comment:
To simplify things, let's just keep this as `_real_extract`. We can use `self._BASE_URL` where needed and add a `_IMPERSONATE_TARGET` class var as well
```suggestion
    def _real_extract(self, url, video_id):
```